aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -57,7 +57,7 @@
     Preconditions.checkNotNull(location, "Location cannot be null.");
 
     this.location = location;
-    String [] schemeSplit = location.split(SCHEME_DELIM);
+    String [] schemeSplit = location.split(SCHEME_DELIM, -1);
```

Comment:
I'm trying to understand the reason for these (and below split changes).  Is this relevant to the larger PR?  My understanding is that this really doesn't change the behavior.

Reply:
It's not, I am just fixing errorprone warnings along the way. Let me remove them since this PR seems pretty big at this point of time.